### ISSUE TITLE
_get_coord_info_new() not compatible with matplotlib 3.9

### ISSUE THREAD

**Issue by maederan201** [qutip/qutip#2535]
### Bug Description

Incompatibility with qutip and matplotlib 3.9 in the matrix_histogram. The histogram cannot be rendered due to a missing argument in the _get_coord_info function.

### Code to Reproduce the Bug

```shell
import matplotlib.pyplot as plt
from qutip.visualization import matrix_histogram
import numpy as np

fig, ax = plt.subplots(subplot_kw={"projection": "3d"})

rho = np.eye(4)

matrix_histogram(rho, ax=ax)

fig.savefig("MWE.pdf")
```

### Code Output

```shell
Traceback (most recent call last):
  File "/tmp/MWE.py", line 11, in <module>
    fig.savefig("MWE.pdf")
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/figure.py", line 3395, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/backends/backend_qtagg.py", line 75, in print_figure
    super().print_figure(*args, **kwargs)
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2204, in print_figure
    result = print_method(
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2054, in <lambda>
    print_method = functools.wraps(meth)(lambda *args, **kwargs: meth(
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/backends/backend_pdf.py", line 2799, in print_pdf
    self.figure.draw(renderer)
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/artist.py", line 95, in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/artist.py", line 72, in draw_wrapper
    return draw(artist, renderer)
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/figure.py", line 3162, in draw
    mimage._draw_list_compositing_images(
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/image.py", line 132, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/maederan/.local/lib/python3.10/site-packages/matplotlib/artist.py", line 72, in draw_wrapper
    return draw(artist, renderer)
  File "/home/maederan/.local/lib/python3.10/site-packages/mpl_toolkits/mplot3d/axes3d.py", line 457, in draw
    axis.draw_pane(renderer)
  File "/home/maederan/.local/lib/python3.10/site-packages/mpl_toolkits/mplot3d/axis3d.py", line 436, in draw_pane
    xys, loc = self.active_pane()
  File "/home/maederan/.local/lib/python3.10/site-packages/mpl_toolkits/mplot3d/axis3d.py", line 415, in active_pane
    mins, maxs, tc, highs = self._get_coord_info()
TypeError: _remove_margins.<locals>._get_coord_info_new() missing 1 required positional argument: 'renderer'
```

### Expected Behaviour

This should produce a matrix histogram.

### Your Environment

```shell
QuTiP Version:      5.0.4
Numpy Version:      1.26.4
Scipy Version:      1.14.0
Cython Version:     3.0.11
Matplotlib Version: 3.9.2
Python Version:     3.10.12
Number of CPUs:     24
BLAS Info:          Generic
INTEL MKL Ext:      False
Platform Info:      Linux (x86_64)
```

### Additional Context

This issue was also mentioned in #2400. It happens due to a change in matplotlib/matplotlib#25272 where the renderer argument was removed from `_get_coord_info`.

**Comment by maederan201** [comment]
The fix seems to be rather straight forward in that the `_get_coord_info_new` function in `_remove_margins` needs to be updated to the new matplotlib one. 

I could get on it myself later this week and create a PR

**Comment by Ericgig** [comment]
Thank you for reporting.

The fix would need to support a few versions of matplotlib, not just the latest. We have already a custom `_axes3D` function that is defined differently according to the version, we could do something similar.

Do you want to give it a try?

**Comment by maederan201** [comment]
I will give it a try

**Comment by UMassIonTrappers** [comment]
Has this been resolved? I still get the error.

**Comment by Ericgig** [comment]
It's fixed in Qutip 5.1
